core/src/main/scala/org/bitcoins/core/wallet/utxo/UTXOSpendingInfo.scala

```diff
@@ -67,19 +67,22 @@ sealed abstract class UTXOSpendingInfo {
   * If you over-specify a path, such as giving a condition where none is needed,
   * then the remaining over-specified path will be ignored.
   *
-  * Note that we do not yet support nested conditionals.
-  *
   * For example, if you wanted to spend a ConditionalScriptPubKey(P2PK1, P2PK2)
   * (which looks like OP_IF <P2PK1> OP_ELSE <P2PK2> OP_ENDIF) with the P2PK1 case,
-  * then you would construct a ConditionalSpendingInfo using ConditionTrue as your
-  * ConditionalPath. Otherwise if you wanted to use P2PK2 you would use ConditionFalse.
+  * then you would construct a ConditionalSpendingInfo using conditionTrue as your
+  * ConditionalPath. Otherwise if you wanted to use P2PK2 you would use conditionFalse.
   */
 sealed trait ConditionalPath
 
 object ConditionalPath {
   case object NoConditionsLeft extends ConditionalPath
-  case object ConditionTrue extends ConditionalPath
-  case object ConditionFalse extends ConditionalPath
+  case class ConditionTrue(nextCondition: ConditionalPath)
+      extends ConditionalPath
+  case class ConditionFalse(nextCondition: ConditionalPath)
+      extends ConditionalPath
+
+  val conditionTrue: ConditionalPath = ConditionTrue(NoConditionsLeft)
```

Comment:
I'm not sure if I understand these, are they just helpers for when we have no conditions left? Why do we need to wrap this in `ConditionTrue`/`ConditionFalse`?

Reply:
Better now with the new names?